src/ui/public/timepicker/timepicker.html

```diff
@@ -59,56 +59,83 @@
+                      ng-change="formatRelativeFrom()"
                       class="form-control col-xs-2">
                     </select>
                   </div>
                   <br>
                   <div class="small">
                     <label>
                       <input
-                        ng-model="relative.round"
-                        ng-checked="relative.round"
-                        ng-change="formatRelative()"
+                        ng-model="relative.from.round"
+                        ng-checked="relative.from.round"
+                        ng-change="formatRelativeFrom()"
                         type="checkbox">
-                      round to the {{units[relative.unit]}}
+                      round to the {{units[relative.from.unit]}}
                     </label>
                   </div>
                 </div>
 
                 <div class="kbn-timepicker-section">
                   <label>
-                    To: Now
+                    To:
+                    <span ng-show="relative.to.preview">{{relative.to.preview}}</span>
+                    <span ng-hide="relative.to.preview"><i>Invalid Expression</i></span>
                   </label>
                   <br>
                   <div class="form-group">
-                    <input type="text" disabled class="form-control" value="Now">
+                    <input
+                      required
+                      ng-model="relative.to.count"
+                      ng-change="formatRelativeTo()"
+                      greater-than="-1"
+                      type="number"
+                      class="form-control">
+                  </div>
+                  <div class="form-group">
+                    <select
+                      ng-model="relative.to.unit"
+                      ng-options="opt.value as opt.text for opt in relativeOptions.to"
+                      ng-change="formatRelativeTo()"
+                      class="form-control col-xs-2">
+                    </select>
+                  </div>
+                  <br>
+                  <div class="small">
+                    <label>
+                      <input
+                        ng-model="relative.to.round"
+                        ng-checked="relative.to.round"
+                        ng-change="formatRelativeTo()"
+                        type="checkbox">
+                      round to the {{units[relative.to.unit]}}
+                    </label>
                   </div>
                 </div>
 
                 <div class="kbn-timepicker-section">
                   <label>&nbsp;</label>
                   <br>
                   <div class="form-group">
-                    <button type="submit" class="btn btn-primary kbn-timepicker-go" ng-disabled="!relative.preview">
+                    <button type="submit" class="btn btn-primary kbn-timepicker-go" ng-disabled="!relative.from.preview !relative.to.preview">
```

Comment:
This ng-disabled should be something like `!relative.from.preview || !relative.to.preview`.